src/git-source-provider.ts

```diff
@@ -245,6 +250,13 @@ export async function cleanup(repositoryPath: string): Promise<void> {
   // Remove auth
   const authHelper = gitAuthHelper.createAuthHelper(git)
   await authHelper.removeAuth()
+
+  // always do this last, otherwise other git commands may fail
+  await git
+    .tryConfigUnsetValue('safe.directory', `'${repositoryPath}$'`, true)
```

Comment:
I think that we should not use the `repositoryPath` literally within a regex. I think that we'd need to translate special characters in the path (like `.`). Do we need a start anchor `^` as well?

But - could we avoid all of that and do `git -cgrep.patternType=fixed unset ...` and treat it like a fixed string and not like a regex?